pandas/core/internals/blocks.py

```diff
@@ -2349,11 +2349,21 @@ def external_values(values: ArrayLike) -> ArrayLike:
     proper extension array).
     """
     if isinstance(values, (PeriodArray, IntervalArray)):
+        warnings.warn(
+            "series.values will stop converting tz from dt64tz, interval to object and period to object",
+            FutureWarning,
+            stacklevel=find_stack_level(),
+        )
```

Comment:
Can you make this warning specific to period / interval. In other words, if they have period data, they should just see the message for period.